### PR TITLE
Eth JSON ABI http route

### PR DESCRIPTION
**Pull request by nickeskov**

### CODE REVIEW

File: pkg/api/app_eth.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"github.com/wavesplatform/gowaves/pkg/proto"
+	"github.com/wavesplatform/gowaves/pkg/proto/ethabi"
+	"github.com/wavesplatform/gowaves/pkg/ride"
+	"github.com/wavesplatform/gowaves/pkg/state"
+)
+
+func (a *App) EthereumDAppMethods(addr proto.WavesAddress) (ethabi.MethodsMap, error) {
+	scriptInfo, err := a.state.ScriptInfoByAccount(proto.NewRecipientFromAddress(addr))
+	if err != nil {
+		if state.IsNotFound(err) {
```
Comment (alexeykiselev) [inline pkg/api/app_eth.go:14]: Why use this method? Is it no enough to use `errors.Is()`? This method `IsNotFound` doesn't use `errors.Is` inside, but checks only "upper" type of the error, doesn't look inside.

File: pkg/api/node_api.go [hunk 2]
```diff
@@ -458,20 +457,52 @@ func (a *NodeApi) stateHash(w http.ResponseWriter, r *http.Request) error {
...
+			id,
+		),
+	)
+}
+
+func (a *NodeApi) EthereumDAppABI(w http.ResponseWriter, r *http.Request) error {
+	s := chi.URLParam(r, "address")
+	addr, err := proto.NewAddressFromString(s)
+	if err != nil {
+		if invalidRune := findFirstInvalidRuneInBase58String(s); invalidRune != nil {
```
Comment (alexeykiselev) [inline pkg/api/node_api.go:474]: Why for report the invalid Base58 rune? Is it not enough to report that the address is invalid? I think that the `findFirstInvalidRuneInBase58String` is highly non-optimal, why not just lookup for chars from Base58 alphabet?

**Review by alexeykiselev — CHANGES_REQUESTED** [review]

File: pkg/api/node_api.go (hunk 2 above)
Comment (nickeskov) [reply pkg/api/node_api.go:474]: Because we have to support scala's errors. Ok, I'll optimize it.

**Review by alexeykiselev — APPROVED** [review]

File: pkg/api/app_eth.go (hunk 1 above)
Comment (nickeskov) [reply pkg/api/app_eth.go:14]: State errors were refactored in last two commits.

**Pull request merged**